pkg/bench/put.go

```diff
@@ -70,7 +70,18 @@ func (u *Put) Start(ctx context.Context, wait chan struct{}) (Operations, error)
 		go func(i int) {
 			rcv := c.Receiver()
 			defer wg.Done()
+
+			// Copy usermetadata and usertags per concurrent thread.
 			opts := u.PutOpts
+			opts.UserMetadata = make(map[string]string, len(u.PutOpts.UserMetadata))
+			opts.UserTags = make(map[string]string, len(u.PutOpts.UserTags))
+			for k, v := range u.PutOpts.UserMetadata {
+				opts.UserMetadata[k] = v
+			}
+			for k, v := range u.PutOpts.UserTags {
+				opts.UserTags[k] = v
+			}
```

Comment:
Not sure why this was causing a problem. I would guess that `UserMetadata` and `UserTags` are read-only field and are not written. The shared map should be fine. 

If the maps are altered from outside `Put.Start`, then it may be safer to perform this copy operation outside the go-routine.

Reply:
It must be copied per goroutine before Put.

Reply:
Which is what I am doing.

Reply:
There is a write happening because the user metadata is used perhaps in MinIO-go to add other metadata (default) per request it can't be shared.

Reply:
I thought that the caller of this method would change the map. In that case it should be put outside the `go` routine on line 70. But if it's changed later, then this is fine.